R/zchunk_L114.bcoc_en_R_S_T_Y.R

```diff
@@ -38,43 +41,112 @@ module_emissions_L114.bcoc_en_R_S_T_Y_DISABLED <- function(command, ...) {
+      mutate(scaled_emissions = scaled_emissions * CONV_KG_TO_TG)
+    # in the old data system, the Bahamas had a missing value for the iso, so their emissions data were dropped
+    if(OLD_DATA_SYSTEM_BEHAVIOR){
+      BCOC_emissions_RCP <- subset( BCOC_emissions_RCP, iso != "bhs")
+    }
+    BCOC_emissions_RCP <- BCOC_emissions_RCP %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") %>%
+      group_by(GCAM_region_ID, Non.CO2, RCP_agg_sector) %>%
+      summarise(scaled_emissions = sum(scaled_emissions)) %>%
+      ungroup()
+
+    # Join the scaled emissions totals from the RCP inventory back into the table of unscaled emissions to compute scalers
+    BCOC_emissions_scaler <- BCOC_unscaled_emissions_RCP %>%
+      left_join_error_no_match(BCOC_emissions_RCP, by = c("GCAM_region_ID", "Non.CO2", "RCP_agg_sector")) %>%
+      mutate(scaler = scaled_emissions / unscaled_emissions) %>%
+      select(GCAM_region_ID, Non.CO2, RCP_agg_sector, scaler)
+
+    # Multiply these scalers by the unscaled emissions by GCAM region, sector, and technology to compute scaled emissions
+    BCOC_scaled_emissions <- BCOC_unscaled_emissions %>%
+      left_join_error_no_match(BCOC_emissions_scaler, by = c("GCAM_region_ID", "Non.CO2", "RCP_agg_sector")) %>%
+      mutate(input.emissions = unscaled_emissions * scaler) %>%
+      # Join in the supplysector/subsector/technology from the mapping table
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, fuel, technology, supplysector, subsector, stub.technology),
+                               by = c("sector", "fuel", "technology")) %>%
+      # Group by categories for GCAM and aggregate the emissions
+      group_by(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology) %>%
+      summarise(input.emissions = sum(input.emissions)) %>%
+      ungroup()
+
+    # Compile energy consumption by the corresponding technologies in order to compute emissions coefficients
+    BCOC_drivers_GCAMtech <- BCOC_drivers %>%
+      filter(Non.CO2 == "BC") %>% #we only need one of the two
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, fuel, technology, supplysector, subsector, stub.technology),
+                                by = c("sector", "fuel", "technology")) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology) %>%
+      summarise(energy = sum(energy)) %>%
+      ungroup()
+
+    #Compute the emissions coefficients for the year 2000
+    L114.bcoc_tgej_R_en_S_F_2000 <- BCOC_scaled_emissions %>%
+      left_join_error_no_match(BCOC_drivers_GCAMtech,
+                                by = c("GCAM_region_ID", "supplysector", "subsector", "stub.technology")) %>%
+      mutate(emissions.factor = input.emissions / energy) %>%
+      na.omit() %>%
+      mutate(year = 2000) %>%
+      select(GCAM_region_ID, Non.CO2, supplysector, subsector, stub.technology, year, emissions.factor) %>%
+      spread(year, emissions.factor)
```

Comment:
We don't need to--I was just trying to make it identical to the existing data table, because the downstream chunk that uses it has already been processed/merged. If this is going to be passed in a different format, then changes to that L151.ctrl chunk will need to be made. But I could do that if you want.